common/common.cpp

```diff
@@ -2361,10 +2361,22 @@ static bool llama_download_file(const std::string & url, const std::string & pat
         curl_easy_setopt(curl.get(), CURLOPT_HEADERFUNCTION, static_cast<CURLOPT_HEADERFUNCTION_PTR>(header_callback));
         curl_easy_setopt(curl.get(), CURLOPT_HEADERDATA, &headers);
 
-        CURLcode res = curl_easy_perform(curl.get());
-        if (res != CURLE_OK) {
-            fprintf(stderr, "%s: curl_easy_perform() failed: %s\n", __func__, curl_easy_strerror(res));
-            return false;
+        int head_request_attempts = 3;
+        int remaining_request_attempts = 3;
+        int retry_delay = 2;
```

Comment:
Value of `head_request_attempts` and `retry_delay` are fixed, you can extract it to a `#define`, for example:

```c
#define CURL_MAX_RETRY 3
#define CURL_RETRY_DELAY_SECONDS 2
```


```suggestion
        int remaining_request_attempts = CURL_MAX_RETRY;
```